lib/Remotery.c

```diff
@@ -8820,4 +8817,92 @@ RMT_API void _rmt_EndMetalSample(void)
 
 #endif // RMT_USE_METAL
 
+// Iterator
+RMT_API rmtSampleIterator _rmt_IterateChildren(rmtSample* sample)
+{
+    rmtSampleIterator iterator;
+    iterator.sample = 0;
+    iterator.initial = sample != NULL ? sample->first_child : 0;
+    return iterator;
+}
+
+RMT_API rmtBool _rmt_IterateNext(rmtSampleIterator* iter)
+{
+    if (iter->initial != NULL)
+    {
+        iter->sample = iter->initial;
+        iter->initial = 0;
+    }
+    else
+    {
+        if (iter->sample != NULL)
+            iter->sample = iter->sample->next_sibling;
+    }
+
+    return iter->sample != NULL ? RMT_TRUE : RMT_FALSE;
+}
+
+// Sample tree accessors
+RMT_API const char* _rmt_SampleTreeGetThreadName(rmtMsgSampleTree* sample_tree)
+{
+    return sample_tree->threadName;
+}
+
+RMT_API rmtSample* _rmt_SampleTreeGetRootSample(rmtMsgSampleTree* sample_tree)
+{
+    return sample_tree->rootSample;
+}
```

Comment:
At first, I passed the `threadName` explicitly to the callback function, but later thought that passing the opaque message pointer fits the api a bit better:
* Easier to add more accessor functions later
* Matches the Sample accessor api used in the callback api

Reply:
Hmmm... yes... that's a good counter argument. I'll edit my comment.